### PR TITLE
[feat]: [DOC-3356]: Adding documentation for the strategy expressions to point to specific strategy levels

### PR DESCRIPTION
**Pull request by brijeshdhakar**
… to point to specific strategy levels

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by archit-harness — DISMISSED** [review]

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 1]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
 
 This expression returns the `identifierPostFix` of the step group when the current node is under the step group, or when the current node is the step group itself, and that step group is a child of a looping strategy.
 
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.identifierPostFix>
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1415]: @brijeshdhakar what is identifierpostfix?

Comment (brijeshdhakar) [reply docs/platform/12_Variables-and-Expressions/harness-variables.md:1415]: Its defined in this class above the current change. 
### identifierPostFix overview, Line:1358

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 2]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+Similarly, you can use other strategy expressions for any specific strategy level if a looping strategy is configured for both the parent and child nodes.
+
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.*>
+By following the provided format, you can retrieve the values of any strategy expressions associated with looping strategies at various levels. This is especially useful when looping strategies are configured within nested levels. Let's take a look at some examples:
+```
+<+strategy.node.sg1.iteration>: Retrieves the current iteration for the node with the identifier sg1(Parent stepGroup).
+<+strategy.node.sg2.iteration>: Retrieves the current iteration for the node with the identifier sg2(Child stepGroup).
+<+strategy.node.some_node_with_looping_strategy.iteration>: Retrieves the current the iteration for the node with identifier some_node_with_looping_strategy.(some_node_with_looping_strategy can be any type of node stage/step/stepGroup)
+
+<+strategy.node.sg1.iterations>: Retrieves the total iterations for the node with the identifier sg1(Parent stepGroup).
```
Comment (SudarshanPurohit) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1434]: What do we mean by "total iterations"? Is it an array of all the iterations, or is it some kind of summary?

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 3]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
 
 This expression returns the `identifierPostFix` of the step group when the current node is under the step group, or when the current node is the step group itself, and that step group is a child of a looping strategy.
 
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.identifierPostFix>
+
+This expressions retrieves the `identifierPostFix` for the node that is child of looping strategy with identifier `STRATEGY_NODE_IDENTIFIER`
+For example, let's consider a scenario with two nested StepGroups: sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. In this case, using the expression `<+stepGroup.identifierPostFix>` would always retrieve the `identifierPostFix` of sg2. To obtain the `identifierPostFix` for specific StepGroup, you can use the following expressions:
```
Comment (SudarshanPurohit) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1418]: Can we add a diagram or screenshot here please? Users will want to refer back to it as they work through the examples below.

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 4]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
 
 This expression returns the `identifierPostFix` of the step group when the current node is under the step group, or when the current node is the step group itself, and that step group is a child of a looping strategy.
 
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.identifierPostFix>
+
+This expressions retrieves the `identifierPostFix` for the node that is child of looping strategy with identifier `STRATEGY_NODE_IDENTIFIER`
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1417]:
```suggestion
This expression retrieves the `identifierPostFix` for the node that is the child of a looping strategy with the identifier `STRATEGY_NODE_IDENTIFIER`.
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md (hunk 3 above)
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1418]:
```suggestion
For example, let's consider two nested step groups, sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. Using the expression, `<+stepGroup.identifierPostFix>` always retrieves the `identifierPostFix` of sg2. To obtain the `identifierPostFix` for a specific step group, you can use the following expressions:
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 5]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+For example, let's consider a scenario with two nested StepGroups: sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. In this case, using the expression `<+stepGroup.identifierPostFix>` would always retrieve the `identifierPostFix` of sg2. To obtain the `identifierPostFix` for specific StepGroup, you can use the following expressions:
+
+```
+<+strategy.node.sg1.identifierPostFix>: Retrieves the identifierPostFix for the node with the identifier sg1(Parent stepGroup).
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1421]:
```suggestion
`<+strategy.node.sg1.identifierPostFix>`: Retrieves the `identifierPostFix` for the node with the identifier sg1 (parent step group).
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 6]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
 
 This expression returns the `identifierPostFix` of the step group when the current node is under the step group, or when the current node is the step group itself, and that step group is a child of a looping strategy.
 
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.identifierPostFix>
+
+This expressions retrieves the `identifierPostFix` for the node that is child of looping strategy with identifier `STRATEGY_NODE_IDENTIFIER`
+For example, let's consider a scenario with two nested StepGroups: sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. In this case, using the expression `<+stepGroup.identifierPostFix>` would always retrieve the `identifierPostFix` of sg2. To obtain the `identifierPostFix` for specific StepGroup, you can use the following expressions:
+
+```
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1420]:
```suggestion

```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 7]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+<+strategy.node.sg1.identifierPostFix>: Retrieves the identifierPostFix for the node with the identifier sg1(Parent stepGroup).
+<+strategy.node.sg2.identifierPostFix>: Retrieves the identifierPostFix for the node with the identifier sg2(Child stepGroup).
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1422]:
```suggestion
`<+strategy.node.sg2.identifierPostFix>`: Retrieves the `identifierPostFix` for the node with the identifier sg2 (child step group).
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 8]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+```
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1423]:
```suggestion

```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 9]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+
+Similarly, you can use other strategy expressions for any specific strategy level if a looping strategy is configured for both the parent and child nodes.
+
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.*>
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1427]:
```suggestion
### <+strategy.node.STRATEGY_NODE_IDENTIFIER.*>

```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 10]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+By following the provided format, you can retrieve the values of any strategy expressions associated with looping strategies at various levels. This is especially useful when looping strategies are configured within nested levels. Let's take a look at some examples:
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1428]:
```suggestion
Using this format, you can retrieve the values of any strategy expressions associated with looping strategies at various levels. This is useful when looping strategies are configured within nested levels. 

Here are some examples:
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 11]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+```
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1429]:
```suggestion

```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 12]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+<+strategy.node.sg1.iteration>: Retrieves the current iteration for the node with the identifier sg1(Parent stepGroup).
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1430]:
```suggestion
* `<+strategy.node.sg1.iteration>`: Retrieves the current iteration of the node with the identifier sg1 (parent step group).
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 13]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
...
+<+strategy.node.sg2.iteration>: Retrieves the current iteration for the node with the identifier sg2(Child stepGroup).
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1431]:
```suggestion
* `<+strategy.node.sg2.iteration>`: Retrieves the current iteration of the node with the identifier sg2 (child step group).
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 14]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
 
 This expression returns the `identifierPostFix` of the step group when the current node is under the step group, or when the current node is the step group itself, and that step group is a child of a looping strategy.
 
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.identifierPostFix>
+
+This expression retrieves the `identifierPostFix` for the node that is the child of a looping strategy with the identifier `STRATEGY_NODE_IDENTIFIER`.
+For example, let's consider two nested step groups, sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. Using the expression, `<+stepGroup.identifierPostFix>` always retrieves the `identifierPostFix` of sg2.
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1418]:
```suggestion
For example, let's consider two nested step groups, sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. The expression, `<+stepGroup.identifierPostFix>` always retrieves the `identifierPostFix` of sg2. 
```

File: docs/platform/12_Variables-and-Expressions/harness-variables.md [hunk 15]
```diff
@@ -1412,6 +1412,34 @@ This expression retrieves the `identifierPostFix` of the stage when the current
 
 This expression returns the `identifierPostFix` of the step group when the current node is under the step group, or when the current node is the step group itself, and that step group is a child of a looping strategy.
 
+### <+strategy.node.STRATEGY_NODE_IDENTIFIER.identifierPostFix>
+
+This expression retrieves the `identifierPostFix` for the node that is the child of a looping strategy with the identifier `STRATEGY_NODE_IDENTIFIER`.
+For example, let's consider two nested step groups, sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. The expression, `<+stepGroup.identifierPostFix>` always retrieves the `identifierPostFix` of sg2. 
+
+To obtain the `identifierPostFix` for a specific step group, you can use the following expressions:
```
Comment (roshnisarangadharan) [inline docs/platform/12_Variables-and-Expressions/harness-variables.md:1420]:
```suggestion
Use the following expressions to obtain the `identifierPostFix` for a specific step group:
```

**Review by roshnisarangadharan — APPROVED** [review]

**Pull request merged**